src/platform/Darwin/BlePlatformDelegateImpl.mm

```diff
@@ -98,7 +98,15 @@
             return found;
         }
 
-        bool BlePlatformDelegateImpl::CloseConnection(BLE_CONNECTION_OBJECT connObj) { return true; }
+        bool BlePlatformDelegateImpl::CloseConnection(BLE_CONNECTION_OBJECT connObj)
+        {
+            CBPeripheral * peripheral = (CBPeripheral *) connObj;
+            // CoreBluetooth API requires a CBCentralManager to close a connection which is a property of the peripheral.
+            CBCentralManager * manager = (CBCentralManager *) [peripheral valueForKey:@"manager"];
```

Comment:
will the key `manager` always exist or should we do a null check?

Are we implementing  `centralManager:didDisconnectPeripheral:error:` and should the return code of true/false be tied to the error code on that callback? 

Why is CloseConnection returning true/false rather than void? Just for logging purposes or is there additional value of knowing close connection success/failure?

Reply:
Grepping through existing code, I believe at some point we should just log errors in the BT stack and make CloseConnection return void. BLEEndpoint seems to be the only one calling CloseConnection and it ignores the return value there, so any errors seem to be silent. I would much rather have the platform BLE layer ensure errors are logged since some calls seem async anyway.

Reply:
Regarding centralManager:didDisconnectPeripheral:error: nobody implements it because the cross platform code already nulls out every related variable at this point so the callback would not be useful.

Reply:
I agree with the possible improvements in the call chain and elsewhere, this PR was only meant to solve one particular issue. There are some pedantic issues which I would not worry about too much, then there are real issues like the following:

uint16_t BlePlatformDelegateImpl::GetMTU(BLE_CONNECTION_OBJECT connObj) const { return 0; }

That should return a real value.

Reply:
Another major issue is the fact that ARC is not enabled. Checking Slack reports stability seems to be a major problem.

Reply:
Added nil check as requested. Btw some code uses NULL, other uses nil. The latter is correct for objc.